code/http/url/url.php

```diff
@@ -387,14 +382,14 @@ public function setPath($path)
         if (is_string($path))
         {
             if (!empty($path)) {
-                $path = explode('/', $path);
+                $path = explode('/', trim($path, '/'));
```

Comment:
@johanjanssens A trim might not be a good idea here. You cannot set the path to foo/bar/ with a trailing slash.